src/sage/combinat/posets/lattices.py

```diff
@@ -734,7 +734,9 @@ def LatticePoset(data=None, *args, **options):
                 error.x = P._vertex_to_element(error.x)
                 error.y = P._vertex_to_element(error.y)
                 raise
-    return FiniteLatticePoset(P, category=FiniteLatticePosets(), facade=P._is_facade)
+    cat = Posets().or_subcategory(options.get('category', None))
```

Comment:
```suggestion
    From sage.categories.posets import Posets
    cat = Posets().or_subcategory(options.get('category', None))
```
Sorry, I forgot to include that this needed to be imported.